zipkin4net-owin/Criteo.Profiling.Tracing.Middleware/ZipkinMiddleware.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Criteo.Profiling.Tracing.Transport;
+using Microsoft.Owin;
+using System.Threading.Tasks;
+
+namespace Criteo.Profiling.Tracing.Middleware
+{
+    class ZipkinMiddleware : OwinMiddleware
+    {
+        private readonly string serviceName;
+        private readonly ITraceExtractor traceExtractor;
+
+        public ZipkinMiddleware(
+            OwinMiddleware next,
+            string serviceName,
+            ITraceExtractor traceExtractor) : base(next)
+        {
+            this.serviceName = serviceName;
+            this.traceExtractor = traceExtractor;
+        }
+
+        public override async Task Invoke(IOwinContext context)
+        {
+            Trace trace;
+
+            if (!this.traceExtractor.TryExtract(context.Request.Headers, (dic, k) => string.Join(",", dic[k]), out trace))
+            {
+                trace = Trace.Create();
+            }
+
+            Trace.Current = trace;
+
+            using (var serverTrace = new ServerTrace(this.serviceName, context.Request.Method))
+            {
+                trace.Record(Annotations.Tag("http.host", context.Request.Host.Value));
+                trace.Record(Annotations.Tag("http.uri", context.Request.Uri.AbsoluteUri));
```

Comment:
No problem but should be aligned with [ASP.NET Core middleware](https://github.com/criteo/zipkin4net/blob/master/zipkin4net-aspnetcore/Criteo.Profiling.Tracing.Middleware/TracingMiddleware.cs) too